### PR TITLE
[CI] Don't compile the Python interfaces on some platforms

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: .github/workflows/meson.yml [hunk 1]
```diff
@@ -171,18 +172,25 @@ jobs:
...
+          fi
+          if [[ ( "${{matrix.compiler}}" == "intel-classic" && "${{matrix.os}}" == "windows-latest" ) || ( "${{matrix.compiler}}" == "intel" && "${{matrix.os}}" == "windows-latest" ) || ( "${{matrix.compiler}}" == "intel-classic" && "${{matrix.os}}" == "macos-latest" ) ]]; then
+            PYTHON_INTERFACE="false"
+          fi
           if [[ "${{matrix.compiler}}" == "nvidia-hpc" || "${{matrix.compiler}}" == "intel-classic" || ( "${{matrix.compiler}}" == "intel" && "${{matrix.os}}" == "windows-latest" ) ]]; then
             SSIDS="false"
           fi
 
           meson setup builddir --buildtype=debug \
+                               --default_library=${SHARED_STATIC} \
```
Comment (amontoison) [inline .github/workflows/meson.yml:186]:
```suggestion
                               --default-library=${SHARED_STATIC} \ 
```

File: .github/workflows/meson.yml [hunk 2]
```diff
@@ -171,18 +172,25 @@ jobs:
...
+          fi
+          if [[ ( "${{matrix.compiler}}" == "intel-classic" && "${{matrix.os}}" == "windows-latest" ) || ( "${{matrix.compiler}}" == "intel" && "${{matrix.os}}" == "windows-latest" ) || ( "${{matrix.compiler}}" == "intel-classic" && "${{matrix.os}}" == "macos-latest" ) ]]; then
+            PYTHON_INTERFACE="false"
+          fi
           if [[ "${{matrix.compiler}}" == "nvidia-hpc" || "${{matrix.compiler}}" == "intel-classic" || ( "${{matrix.compiler}}" == "intel" && "${{matrix.os}}" == "windows-latest" ) ]]; then
             SSIDS="false"
           fi
 
-          meson setup builddir --buildtype=debug \
+          meson setup builddir --default-library=${SHARED_STATIC} \
```
Comment (amontoison) [inline .github/workflows/meson.yml:185]:
```suggestion
          meson setup builddir -Ddefault_library=${SHARED_STATIC} \ 
```

File: .github/workflows/meson.yml [hunk 3]
```diff
@@ -171,18 +172,25 @@ jobs:
...
+          fi
+          if [[ ( "${{matrix.compiler}}" == "intel-classic" && "${{matrix.os}}" == "windows-latest" ) || ( "${{matrix.compiler}}" == "intel" && "${{matrix.os}}" == "windows-latest" ) || ( "${{matrix.compiler}}" == "intel-classic" && "${{matrix.os}}" == "macos-latest" ) ]]; then
+            PYTHON_INTERFACE="false"
+          fi
           if [[ "${{matrix.compiler}}" == "nvidia-hpc" || "${{matrix.compiler}}" == "intel-classic" || ( "${{matrix.compiler}}" == "intel" && "${{matrix.os}}" == "windows-latest" ) ]]; then
             SSIDS="false"
           fi
 
-          meson setup builddir --buildtype=debug \
+          meson setup builddir -Ddefault_library=${SHARED_STATIC} \
```
Comment (amontoison) [inline .github/workflows/meson.yml:185]:
```suggestion
          meson setup builddir -Ddefault_library=${SHARED_STATIC} \
```

**Comment by amontoison** [comment]
@nimgould 
I update CI to not compile the Python interfaces on some platforms (Mac + Intel / Windows + Intel).
I also modified the options to compile static libraries on Windows instead of shared ones with Intel compilers.

We have a compilation error for a test:
https://github.com/ralna/GALAHAD/actions/runs/7809845943/job/21302316696?pr=247#step:12:6950

**Comment by amontoison** [comment]
Superseded by #248

**Comment by nimgould** [comment]
I have no idea, this works fine under linux. But, I don't see an instruction (-fpp) to run the preprocessor in the compile line.

**Pull request closed without merging**